lte/gateway/python/magma/mobilityd/subscriberdb_client.py

```diff
@@ -60,3 +61,48 @@ def get_subscriber_ip(self, sid: str) -> ipaddress.ip_address:
                 err.code(),
                 err.details())
             return None
+
+    def get_subscriber_apn_vlan(self, sid: str) -> str:
+        """
+        Make RPC call to 'GetSubscriberData' method of local SubscriberDB
+        service to get assigned IP address if any.
+        TODO: Move this API to separate APN configuration service.
+        """
+        if self.subscriber_client is None:
+            return None
+
+        ip_addr = None
+        try:
+            if '.' in sid:
+                imsi, apn_name = sid.split('.', maxsplit=1)
+            else:
+                imsi, apn_name = sid, ''
+
+            if apn_name in self.apn_vlan_map:
+                return self.apn_vlan_map[apn_name]
+
+            data = self.subscriber_client.GetSubscriberData(SIDUtils.to_pb(imsi))
+            if data and data.non_3gpp and data.non_3gpp.apn_config:
+                vlan_id = None
+                for apn_config in data.non_3gpp.apn_config:
+                    if apn_config.service_selection == '*':
+                        vlan_id = apn_config.vlan_id
+                    if apn_config.service_selection == apn_name:
+                        vlan_id = apn_config.vlan_id
+                        break
+
+                if vlan_id is not None:
+                    self.apn_vlan_map[apn_name] = str(int(vlan_id))
```

Comment:
why aren't you sending the vlan ID down as an int in the first place and keeping it that way?

Reply:
ok, I can change it to int.